app/models/work.rb

```diff
@@ -85,16 +85,25 @@ def administered_by?(user)
   end
 
   class << self
-    def unfinished_works(user)
-      works_by_user_state(user, ["none", "draft", "awaiting_approval"])
+    def unfinished_works(user, search_terms = nil)
+      works = works_by_user_state(user, ["none", "draft", "awaiting_approval"])
+      filter_works(works, search_terms)
     end
 
-    def completed_works(user)
-      works_by_user_state(user, "approved")
+    def completed_works(user, search_terms = nil)
+      works = works_by_user_state(user, "approved")
+      filter_works(works, search_terms)
     end
 
-    def withdrawn_works(user)
-      works_by_user_state(user, "withdrawn")
+    def withdrawn_works(user, search_terms = nil)
+      works = works_by_user_state(user, "withdrawn")
+      filter_works(works, search_terms)
+    end
+
+    def filter_works(works, search_terms)
+      return works if search_terms.nil?
+      terms = search_terms.strip.downcase
```

Comment:
This will become slow if the user's work list is large.  Likely that is why they are searching.

It would be better to send the query to the database.  I believe a like query can be done on a JSONB field.

Reply:
Agree that this should probably move to the database, eventually, but for a slightly different reason: When we have enough records that query performance becomes a problem, we will need to page the results, and we would need to filter before paging.

Unless paging has already been implemented, I don't know that this is essential, but it wouldn't be bad to tackle it now, either.

Reply:
Moved the search to the database.